### PR TITLE
chore: List public API

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates to SAP/cloud-sdk-backlog#406.

I made a first list of methods and object which I think are useful for external consumer. I would also like to discuss the general strategy. Currently everything which is needed for compiling is exported. However, this includes methods which are not useful as public methods.

We have discussed to have a public API -> reuse for customers
An `internal` file containing all exports und `@sap-cloud-sdk/package/internal`

This PR is for listing things we want to put in our public API. In the `.md` files I made a first suggestion. Please review and discuss. The files can be understood like

function myFunction
   interface functionArgument
      type typeUsedInInterface...

### CODE REVIEW

**Comment by FrankEssenberger** [comment]
Discussion: https://github.com/SAP/cloud-sdk-js/pull/1750

File: packages/openapi/public-api.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+### openApi Package Public API
+
+-
```
Comment (deekshas8) [inline packages/openapi/public-api.md:3]:
```suggestion
- class `OpenApiRequestBuilder`
```

File: packages/odata-v4/public-api.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+### oData-v4 Package Public API
+
+- all filterFunctions new to v4 (contains,matchesPattern,fractionalSeconds,totalOffsetMinutes,maxDateTime,minDateTime,now,hasSubset,hasSubsequence)
+- function any(),all()
+  - Class FilterLambdaExpression
+    -build() (on function imports)
```
Comment (jjtang1985) [inline packages/odata-v4/public-api.md:6]:
```suggestion
  - Class FilterLambdaExpression
    - build() (on function imports)
```

File: packages/odata-v2/public-api.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+### oData-v2 Package Public API
+
+- all filterFunctions new to v2 (substringOf,replace)
+  -build() (on function imports)
```
Comment (jjtang1985) [inline packages/odata-v2/public-api.md:4]:
```suggestion
- all filterFunctions new to v2 (substringOf,replace)
  - build() (on function imports)
```

File: packages/generator-common/public-api.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+### Generator-Common Package Public API
+
+Public API is empty => this should get reimported on generator and openapi-generator
+
+- function transpileDirectory()
+- function readCompilerOptions()
```
Comment (jjtang1985) [inline packages/generator-common/public-api.md:6]: Maybe keep this list empty and move to the generator packages?

**Review by jjtang1985 — APPROVED** [review]
LGTM as initial list.
Do we have a process for maintenance?

File: packages/generator-common/public-api.md (hunk 4 above)
Comment (FrankEssenberger) [reply packages/generator-common/public-api.md:6]: I leave a comment

**Pull request merged**